utils/MerkleTree.ts

```diff
@@ -1,145 +1,2 @@
-// This script provides some useful methods for building MerkleTrees. It is essentially the uniswap implementation
-// https://github.com/Uniswap/merkle-distributor/blob/master/src/merkle-tree.ts with some added convenience methods
-// to take the leaves and conversion functions, so the user never has to work with buffers.
-import { bufferToHex, keccak256 } from "ethereumjs-util";
-
+export { MerkleTree } from "@uma/common";
 export const EMPTY_MERKLE_ROOT = "0x0000000000000000000000000000000000000000000000000000000000000000";
-export class MerkleTree<T> {
-  private readonly elements: Buffer[];
-  private readonly bufferElementPositionIndex: { [hexElement: string]: number };
-  private readonly layers: Buffer[][];
-
-  constructor(leaves: T[], public readonly hashFn: (element: T) => string) {
-    this.elements = leaves.map((leaf) => this.leafToBuf(leaf));
-    // Sort elements
-    this.elements.sort(Buffer.compare);
-    // Deduplicate elements
-    this.elements = MerkleTree.bufDedup(this.elements);
-
-    this.bufferElementPositionIndex = this.elements.reduce<{ [hexElement: string]: number }>((memo, el, index) => {
-      memo[bufferToHex(el)] = index;
-      return memo;
-    }, {});
-
-    // Create layers
-    this.layers = this.getLayers(this.elements);
-  }
-
-  isEmpty(): boolean {
-    return this.layers.length === 0;
-  }
-
-  getLayers(elements: Buffer[]): Buffer[][] {
-    const layers: Buffer[][] = [];
-    if (elements.length === 0) return layers;
-
-    layers.push(elements);
-
-    // Get next layer until we reach the root
-    while (layers[layers.length - 1].length > 1) {
-      layers.push(this.getNextLayer(layers[layers.length - 1]));
-    }
-
-    return layers;
-  }
-
-  getNextLayer(elements: Buffer[]): Buffer[] {
-    return elements.reduce<Buffer[]>((layer, el, idx, arr) => {
-      if (idx % 2 === 0) {
-        // Hash the current element with its pair element
-        layer.push(MerkleTree.combinedHash(el, arr[idx + 1]));
-      }
-
-      return layer;
-    }, []);
-  }
-
-  static combinedHash(first: Buffer, second: Buffer): Buffer {
-    if (!first) {
-      return second;
-    }
-    if (!second) {
-      return first;
-    }
-
-    return keccak256(MerkleTree.sortAndConcat(first, second));
-  }
-
-  getRoot(): Buffer {
-    return this.layers[this.layers.length - 1][0];
-  }
```

Comment:
I think the implementations differ between this one and the one in @uma/common. I think we made some additions to this implementation to handle empty roots. The empty case is handled by returning the zero root (and this one has an "isEmpty()" method, whereas the one in @uma/common throws I think.

Maybe we should update the one in @uma/common before importing in everywhere in across so that it has these extra features. Thoughts?

Reply:
good point https://github.com/UMAprotocol/protocol/pull/4318

Reply:
Updated the @uma/common merkleTree implementation which is now exported in @uma/common:2.29